### PR TITLE
feat(destructive-button): added new destructive button type to button

### LINKED ISSUE #1635: Buttons: add destructive button styles (EVO)
Destructive buttons were not properly documented. They have been built out to have primary, secondary, and tertiary options along with states and sizes. Destructive hover states will use R5 and active will use R6 (maintains the X4 -> X5 -> X6 pattern).

[Documentation](https://www.figma.com/file/zEBdEhbonrBOGzZ0fXzWvM/?node-id=0%3A327)

### PR DESCRIPTION
**Pull request by ArtBlue**
## Description
Adds destructive button type to button component.

## Context
Following the design specifications with regard to look and feel. I kept this as a type of button instead of a separate component since it shared so much of the structural aspects and differed only in colors.

## References
Closes #1635 

## Screenshots
destructive-buttons-skin

### CODE REVIEW

File: src/less/button/base/button.less [hunk 1]
```diff
@@ -171,6 +204,38 @@ a.fake-btn--secondary[href] {
...
+
+button.btn--secondary.btn--destructive,
+a.fake-btn--secondary.fake-btn--destructive {
+    .background-color-token(btn-secondary-destructive-background-color, color-background-default);
+    .border-color-token(btn-secondary-destructive-border-color, color-destructive);
+    .color-token(btn-secondary-destructive-foreground-color, color-destructive);
+}
+
+button.btn--secondary.btn--destructive[disabled] svg.icon {
+    fill: currentColor;
```
Comment (agliga) [inline src/less/button/base/button.less:216]: Why is this needed? I think current color should be coming from icon?

File: src/less/tokens/base/tokens.less [hunk 2]
```diff
@@ -27,6 +27,25 @@
...
+@color-destructive-primary-disabled-background: @color-grey3;
+@color-destructive-secondary-text: @color-r4;
+@color-destructive-secondary-hover-text: @color-r5;
+@color-destructive-secondary-active-text: @color-r6;
+@color-destructive: @color-r4;
+@color-destructive-hover: @color-r5;
+@color-destructive-active: @color-r6;
+
+// destructive spinners
+@color-destructive-primary-spinner-background: #ec7089; // short section
```
Comment (agliga) [inline src/less/tokens/base/tokens.less:42]: Should use a variable for this and define it in primitives.

**Review by agliga — COMMENTED** [review]
Just a glance over the code: really good, looks great
Couple of minor nits/questions. Will try it out locally after the comments are addressed.

File: src/less/button/base/button.less (hunk 1 above)
Comment (ArtBlue) [reply src/less/button/base/button.less:216]: I believe it was necessary, at least at the point at which I was seeing an issue where the color was off. I just removed it and it looks like it works without it. I'll remove it since it appears to be redundant now.

File: src/less/tokens/base/tokens.less (hunk 2 above)
Comment (ArtBlue) [reply src/less/tokens/base/tokens.less:42]: We could. I was debating that but since design did not have established token names, to avoid future confusion, I did not want to start a new convention that may need to be changed later.

**Review by agliga — APPROVED** [review]
Overall awesome work! Looks really great and simple.
